packages/fxa-settings/src/index.tsx

```diff
@@ -39,26 +41,53 @@ try {
     return document.head.querySelector(name);
   });
 
-  const apolloClient = createApolloClient(config.servers.gql.url);
-  const appContext = initializeAppContext();
+  const completeInitialization = () => {
+    const apolloClient = createApolloClient(config.servers.gql.url);
+    const appContext = initializeAppContext();
 
-  render(
-    <React.StrictMode>
-      <AppErrorBoundary>
-        <AppContext.Provider value={appContext}>
-          <AppLocalizationProvider
-            baseDir="/settings/locales"
-            userLocales={navigator.languages}
-          >
-            <ApolloProvider client={apolloClient}>
-              <App {...{ flowQueryParams }} />
-            </ApolloProvider>
-          </AppLocalizationProvider>
-        </AppContext.Provider>
-      </AppErrorBoundary>
-    </React.StrictMode>,
-    document.getElementById('root')
-  );
+    render(
```

Comment:
Do you mean the session token needs to be updated before we perform any queries in Settings? I think we can do that.

Reply:
I'll double-check if it's still the same, but the issue was that on `createApolloClient` if there were no session token the page would trigger an error and redirect the user to the sign-in page

We want to make sure that only happens after the fxa status message is received (so one option is to delay the initialization of the Apollo client, which is what I did here, maybe there is a better way?)

Reply:
> I'll double-check if it's still the same, but the issue was that on createApolloClient if there were no session token the page would trigger an error and redirect the user to the sign-in page

This kind of relates to another ticket I'm working on this sprint (FXA-8817), which a partial fix for is to not send up a GQL request if the session token doesn't exist. I think that would fix this problem.

I'd suggested in https://github.com/mozilla/fxa/pull/15723#discussion_r1435291457 that we do this check inside `App`. Maybe that's a good place for it also given we talked about possibly receiving something like the oauth sync context through fxa_status?

If you fix merge conflicts here I could push something up in a separate commit for you to look at.

Reply:
For posterity I'm bringing some of what I'd said in Slack here and bear with me, I'm mixing PRs/issues since it all relates...

I think FXA-8817 is actually unrelated, I'd started on it and it won't resolve this. However I think when you originally worked on this ticket, we had [this logic](github.com/mozilla/fxa/blob/e3fb98be4aa6b928383fb9dc06c6764c64a68f74/packages/fxa-settings/src/components/App/index.tsx#L175-L178) more tightly coupled with apollo-client and now it's more workable inside `App`.

While this needs more tweaking, I'd started on FXA-8671 and [left a comment here](https://github.com/mozilla/fxa/pull/16209/files#diff-36aa53053f1cbb57d4a94e4cf502a82d89c9ef2cfdb037ffc6e31ad298c74893R112) about where I pictured this going. In my mind, we can use this to receive all of the goodies from the browser:
* `signedInUser` as you are here, which probably partially covers the "merge browser account" part of FXA-8828
* Desktop oauth data if applicable*, which we discussed receiving via web channel instead of params
* Sync engines if applicable

*We'll still need to modify content-server to set `context` conditionally for the "Looking for Firefox Sync?" link which has not yet been Reactified. 

Thinking through this more though, if we're going to have to send down `context` via a web channel couldn't we go ahead and send down the desktop oauth params to place on the link for one less round trip or is there a downside to doing that? 🤔 That would make it more consistent with sync mobile (oauth data being in the query params) and I could modify my other branch accordingly.

The above question shouldn't affect this branch though so let's continue discussion over there. This PR can still be responsible for receiving the event on the React side and the `signedInUser` bit. I was hoping hoping to clean up the sync engine web channel bits that are in `Signup/container.tsx` in that other branch with this web channel message, but I might end up doing that in a follow up.